src/main/scala/prv/zielony/scala/tutorials/PeselValidator.scala

```diff
@@ -6,49 +6,41 @@ package prv.zielony.scala.tutorials
 object PeselValidator {
 
   /**
-    * The aim is to validate the PESEL number, which is more or less a Polish ID number.
-    * The validation should go as follows:
-    * 1. The number has to have 11 digits
-    * 2. The following algorithm should hold:
-    *
-    *   We multiply multiply digits respectively:
-    *   - first by 1
-    *   - second by 3
-    *   - third by 7
-    *   - fourth by 9
-    *   - fifth by 1
-    *   - sixth by 3
-    *   - seventh by 7
-    *   - eight by 9
-    *   - ninth by 1
-    *   - tenth by 3
-    *   - eleventh by 1
-    *   where the first digit is the one on the left-hand side as the number is written, i.e. for
-    *   1234 the first digit is 1 and the fourth is 4.
-    *   Next, we sum all the results. If the final result is divisible by 10, the PESEL number is correct.
-    *   Otherwise, it isn't.
-    */
-
-  def validate(pesel:Long):Boolean = {
-
-    val factors =  List(1, 3, 7, 9, 1, 3, 7, 9, 1, 3, 1)
-
-    var sum: Long = 0;
-
-    for(i <- 0 until factors.size){
-      sum += digit(pesel, 11 - i) * factors(i)
+   * The aim is to validate the PESEL number, which is more or less a Polish ID number.
+   * The validation should go as follows:
+   * 1. The number has to have 11 digits
+   * 2. The following algorithm should hold:
+   *
+   *   We multiply multiply digits respectively:
+   *   - first by 1
+   *   - second by 3
+   *   - third by 7
+   *   - fourth by 9
+   *   - fifth by 1
+   *   - sixth by 3
+   *   - seventh by 7
+   *   - eight by 9
+   *   - ninth by 1
+   *   - tenth by 3
+   *   - eleventh by 1
+   *   where the first digit is the one on the left-hand side as the number is written, i.e. for
+   *   1234 the first digit is 1 and the fourth is 4.
+   *   Next, we sum all the results. If the final result is divisible by 10, the PESEL number is correct.
+   *   Otherwise, it isn't.
+   */
+
+  package prv.zielony.scala.tutorials
+
+  object PeselValidator{
+    def validate(pesel:List[Int]):Boolean={
+      if (pesel.length==11){
+        var multipliers:List[Int] = List(1,3,7,9,1,3,7,9,1,3,1)
+        if ((pesel,multipliers).zipped.map(_*_).sum % 10 == 0) {
```

Comment:
This is a very good solution, but since some people on that stage may yet not know the "map" and "zipped" methods in general, I'd rather we did that Java - style, with a simple loop, if you don't mind